### PR TITLE
EVG-19935: Add SaveDistroSection mutation

### PR DESCRIPTION
**Pull request by minnakt**
EVG-19935

### Description
This PR introduces the SaveDistroSection resolver, which currently only works for the general section. Other sections can be added incrementally. The implementation follows a similar pattern to project settings, in that a subset of fields will be updated per section.

### Testing
- Unit tests &  GraphQL tests

### CODE REVIEW

**Comment by minnakt** [comment]
**Note:** Since the implementation diverged significantly from the technical design, I'm going to wait for feedback from the UI team before requesting an app review, to confirm that this is going on the right track. Tagging Sophie since you have the most context about why I had to make these changes 🙏

File: graphql/mutation_resolver.go
Comment (sophstad) [inline graphql/mutation_resolver.go:129]: I'll defer to an app engineer's opinion on this, but it may be preferable to move the heavy logic out of the resolver/GraphQL utils and into `rest/data/distros.go`

File: graphql/util.go
Comment (sophstad) [inline graphql/util.go:1037]:
```suggestion
func handleDistroOnSaveOperation(ctx context.Context, distroID string, userID string, onSave DistroOnSaveOperation) (int, error) {
```

File: graphql/util.go
Comment (sophstad) [inline graphql/util.go:1071]:
```suggestion
			return 0, InternalServerError.Send(ctx, fmt.Sprintf("marking hosts as needing Jasper service restarted for distro '%s': %s", distroID, err))
```

File: graphql/util.go
Comment (sophstad) [inline graphql/util.go:1068]: Maybe we could just add the host ID to the catcher here since the error is wrapped in a similar message (and same above)

**Review by malikchaya2 — APPROVED** [review]
Approved with some nits

File: graphql/mutation_resolver.go
Comment (khelif96) [reply graphql/mutation_resolver.go:1]: @SupaJoon do you know why this is still happening?

File: validator/distro_validator.go
Comment (sophstad) [inline validator/distro_validator.go:567]: [nit] I would probably grab settings from the Evergreen environment in this function directly, your call though!

File: validator/distro_validator.go
Comment (sophstad) [inline validator/distro_validator.go:554]: Would you mind adding a unit test for this validator?

File: graphql/util.go
Comment (sophstad) [inline graphql/util.go:1001]: This is tricky stuff, nice work Minna 😍

File: graphql/schema/types/distro.graphql
Comment (sophstad) [inline graphql/schema/types/distro.graphql:25]: Not sure if this is a practice worth keeping around, but we do usually name inputs with `Input` at the end
```suggestion
input SaveDistroInput {
```

**Review by sophstad — CHANGES_REQUESTED** [review]

File: graphql/schema/types/distro.graphql
Comment (minnakt) [reply graphql/schema/types/distro.graphql]: oops sorry yes!

File: model/distro/distro.go
Comment (minnakt) [reply model/distro/distro.go:892]: It might be tricky to combine them at this point since only one section (the general section) is complete and there may be some unexpected curve balls down the road -- we could definitely revisit it though!

**Review by sophstad — APPROVED** [review]

**Pull request merged**
